test/object-store/sync/flx_role_change.cpp

```diff
@@ -776,9 +776,8 @@ TEST_CASE("flx: role changes during bootstrap complete successfully", "[sync][fl
     }
     SECTION("Role change during subscription bootstrap") {
         auto realm_1 = Realm::get_shared_realm(config);
-        /// TODO: update to:
-        ///       bool initial_subscription = GENERATE(false, true);
-        bool initial_subscription = true;
+        bool initial_subscription = GENERATE(false, true);
```

Comment:
is this hitting the bug?

Reply:
Yes - the bug was being hit when a role change occurred during the initial subscription (query version 1) and there would be extra data in the database since the pending bootstrap store was not reset when the session restarted.

Even though the permissions were updated to only allow "employee" records, there were still some stale manager and director records in the bootstrap store from when only "managers" and "directors" records were allowed, leading into an incorrect number of entries in the database.

Reply:
Was this always reproducible? Otherwise I think we should add a new test.